Exercicio4/src/script.ts

```diff
@@ -0,0 +1,146 @@
+let lista: Array<Object> = [
+    {"id" : 1, "name": "Ada Lovelace", "bio" : "Ada Lovelace, foi uma matemática e escritora inglesa reconhecida por ter escrito o primeiro algoritmo para ser processado por uma máquina"},
+    {"id" : 2, "name": "Alan Turing", "bio" : "Alan Turing foi um matemático, cientista da computação, lógico, criptoanalista, filósofo e biólogo teórico britânico, ele é amplamente considerado o pai da ciência da computação teórica e da inteligência artificia"},
+    {"id" : 3, "name": "Nikola Tesla", "bio" : "Nikola Tesla foi um inventor, engenheiro eletrotécnico e engenheiro mecânico sérvio, mais conhecido por suas contribuições ao projeto do moderno sistema de fornecimento de eletricidade em corrente alternada."},
+    {"id" : 4, "name": "Nicolau Copérnico", "bio": "Nicolau Copérnico foi um astrônomo e matemático polonês que desenvolveu a teoria heliocêntrica do Sistema Solar."}
+];
+
+let id: string;
+
+const Armazenamento: Object = {
+    buscar(): Array<Object> {
+        return JSON.parse(localStorage.getItem("helpper.origins:cientistas")) || [];
+    },
+
+    gravar(cientistas: Object): void {
+        localStorage.setItem("helpper.origins:cientistas", JSON.stringify(cientistas));
+    }
+}
+
+const Cientista: Object = {
+    editar(): void {
+        // pega os valores dos inputs
+        let inputNome: HTMLInputElement = document.querySelector("#campo-nome");
+        let inputBio: HTMLInputElement = document.querySelector("#campo-bio");
+        let name: string = inputNome.value;
+        let bio: string = inputBio.value;
+        
+        let cientistaEditado: object = {id: Number(id), name, bio};
+        lista = Armazenamento["buscar"]();
```

Comment:
substituir todas as chamadas a propriedades que estão sendo chamadas por chave "[]" pela notação por ponto, por exemplo aqui onde se tem Armazenamento"buscar" substituir por Armazenamento.buscar(), onde se tem obj["id"] substituir por obj.id e assim por diante em toda a classe

Reply:
Tentei fazer dessa forma, mas apareceu o seguinte erro: 
error TS2339: Property 'buscar' does not exist on type 'Object'.

29         lista = Armazenamento.buscar();
                                 ~~~~~~


Found 1 error.